core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala

```diff
@@ -216,26 +219,23 @@ class DruidBeamMaker[A: Timestamper](
 
 object DruidBeamMaker
 {
-  def generateBaseFirehoseId(dataSource: String, segmentGranularity: Granularity, ts: DateTime, partition: Int) = {
+  def generateBaseFirehoseId(dataSource: String,taskDuration:Period, ts: DateTime, partition: Int) = {
     // Not only is this a nasty hack, it also only works if the RT task hands things off in a timely manner. We'd rather
     // use UUIDs, but this creates a ton of clutter in service discovery.
+    //It may also break if task duration is changed across RT tasks
 
     val tsUtc = new DateTime(ts.millis, ISOChronology.getInstanceUTC)
 
-    val cycleBucket = segmentGranularity match {
-      case Granularity.MINUTE => tsUtc.minuteOfHour().get
-      case Granularity.FIVE_MINUTE => tsUtc.minuteOfHour().get
-      case Granularity.TEN_MINUTE => tsUtc.minuteOfHour().get
-      case Granularity.FIFTEEN_MINUTE => tsUtc.minuteOfHour().get
-      case Granularity.HOUR => tsUtc.hourOfDay().get
-      case Granularity.SIX_HOUR => tsUtc.hourOfDay().get
-      case Granularity.DAY => tsUtc.dayOfMonth().get
-      case Granularity.WEEK => tsUtc.weekOfWeekyear().get
-      case Granularity.MONTH => tsUtc.monthOfYear().get
-      case Granularity.YEAR => tsUtc.yearOfCentury().get
-      case x => throw new IllegalArgumentException("No gross firehose id hack for granularity[%s]" format x)
+
+    val cycleBucket = taskDuration match {
+      case i if(i.toStandardSeconds.getSeconds < DateTimeConstants.SECONDS_PER_HOUR) => "%02d".format(tsUtc.minuteOfHour().get)
+      case i if(i.toStandardSeconds().getSeconds < DateTimeConstants.SECONDS_PER_DAY) => "%02d-%02d" .format(tsUtc.hourOfDay().get,tsUtc.minuteOfHour().get)
+      case i if(i.toStandardSeconds().getSeconds < DateTimeConstants.SECONDS_PER_DAY *28) => "%02d-%02d-%02d" .format(tsUtc.dayOfMonth().get,tsUtc.hourOfDay().get,tsUtc.minuteOfHour().get)
+      case i if(i.toStandardSeconds.getSeconds < DateTimeConstants.SECONDS_PER_DAY*365) => "%02d-%02d-%02d-%02d" .format(tsUtc.monthOfYear().get,tsUtc.dayOfMonth().get,tsUtc.hourOfDay().get,tsUtc.minuteOfHour().get)
+      case i if(i.toStandardSeconds.getSeconds > DateTimeConstants.SECONDS_PER_DAY*365) => "%02d-%02d-%02d-%02d-%02d" .format(tsUtc.yearOfCentury().get,tsUtc.monthOfYear().get,tsUtc.dayOfMonth().get,tsUtc.hourOfDay().get,tsUtc.minuteOfHour().get)
```

Comment:
This looks like it's supposed to be a fallback, so it could just be:

```scala
case i => "%02d-%02d-%02d-%02d-%02d" .format(tsUtc.yearOfCentury().get,tsUtc.monthOfYear().get,tsUtc.dayOfMonth().get,tsUtc.hourOfDay().get,tsUtc.minuteOfHour().get)
```